analyzers/tests/SonarAnalyzer.UnitTest/Helpers/TypeDeclarationSyntaxExtensionsTest.cs

```diff
@@ -91,5 +91,148 @@ private void Method2()
+            var typeDeclaration = tree.GetCompilationUnitRoot().DescendantNodesAndSelf().OfType<TypeDeclarationSyntax>().Single();
+            var methodSymbol = typeDeclaration.PrimaryConstructor(compilation.GetSemanticModel(tree));
+            methodSymbol.Should().NotBeNull();
+            methodSymbol.MethodKind.Should().Be(MethodKind.Constructor);
+            methodSymbol.Parameters.Should().ContainSingle();
+            methodSymbol.Parameters.Should().ContainSingle(p => p.Name == "i" && p.Type.SpecialType == SpecialType.System_Int32);
+        }
+
+        [DataTestMethod]
+        [DataRow("class")]
+        [DataRow("struct")]
+        [DataRow("readonly struct")]
+        [DataRow("record struct")]
+#if NET
+        [DataRow("record")]
+        [DataRow("record class")]
+#endif
+        public void PrimaryConstructor_PrimaryConstructorOnClass(string type)
+        {
+            var (tree, model) = TestHelper.CompileCS($$"""
+            {{type}} Test(int i)
+            {
+            }
+            """);
+            var typeDeclaration = tree.GetCompilationUnitRoot().DescendantNodesAndSelf().OfType<TypeDeclarationSyntax>().Single();
+            var methodSymbol = typeDeclaration.PrimaryConstructor(model);
+            methodSymbol.Should().NotBeNull();
+            methodSymbol.MethodKind.Should().Be(MethodKind.Constructor);
+            methodSymbol.Parameters.Should().ContainSingle();
+            methodSymbol.Parameters.Should().ContainSingle(p => p.Name == "i" && p.Type.SpecialType == SpecialType.System_Int32);
+        }
+
+        [TestMethod]
+        public void PrimaryConstructor_EmptyPrimaryConstructor()
+        {
+            var (tree, model) = TestHelper.CompileCS($$"""
+            public class Test()
+            {
+                public Test(int i) : this() { }
+            }
+            """);
+            var typeDeclaration = tree.GetCompilationUnitRoot().DescendantNodesAndSelf().OfType<TypeDeclarationSyntax>().Single();
+            var methodSymbol = typeDeclaration.PrimaryConstructor(model);
+            methodSymbol.Should().NotBeNull();
+            methodSymbol.MethodKind.Should().Be(MethodKind.Constructor);
+            methodSymbol.Parameters.Should().BeEmpty();
+        }
+
+        [TestMethod]
+        public void PrimaryConstructor_EmptyPrimaryConstructorAndStaticConstructor()
+        {
+            var (tree, model) = TestHelper.CompileCS($$"""
+            public class Test()
+            {
+                static Test() { }
+            }
+            """);
+            var typeDeclaration = tree.GetCompilationUnitRoot().DescendantNodesAndSelf().OfType<TypeDeclarationSyntax>().Single();
+            var methodSymbol = typeDeclaration.PrimaryConstructor(model);
+            methodSymbol.Should().NotBeNull();
+            methodSymbol.MethodKind.Should().Be(MethodKind.Constructor);
+            methodSymbol.Parameters.Should().BeEmpty();
+            methodSymbol.IsStatic.Should().BeFalse();
```

Comment:
I don't understand this, shouldn't it retrieve the Test() static ctor?
I also suggest adding some parameters to the primary ctor so it's easier to understand from the test which one are we referring to in the assertions

Reply:
We are looking for a primary constructor. A primary constructor can not be a static constructor. (Don't confuse it with the default constructor).